### PR TITLE
Change mixer to use FileNotFoundError and fix related tests

### PR DESCRIPTION
**Pull request by Temmie3754**
This resolves https://github.com/pygame/pygame/issues/2792 by changing the error emitted and the test checking the error. This also fixes an issue with a test incorrectly failing because "allowedchanges" was not set to 0.

### CODE REVIEW

**Comment by Starbuck5** [comment]
Thanks for this.
The mixer.music documentation should get a `versionchanged` label to reflect this change.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

As Starbuck5 said, just need to add something like:
```
.. versionchanged:: 2.2.0 - Now raises a FileNotFoundError when no file is found, rather than a pygame.error
```
line to `music.rst` (roughly line 37/38 with spacing around it).

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: src_c/music.c [hunk 1]
```diff
@@ -385,7 +385,7 @@ _load_music(PyObject *obj, char *namehint)
     if (rw ==
         NULL) { /* stop on NULL, error already set is what we SHOULD do */
         PyErr_Fetch(&_type, &error, &_traceback);
-        PyErr_SetObject(pgExc_SDLError, error);
+        PyErr_SetObject(PyExc_FileNotFoundError, error);
         Py_XDECREF(_type);
         Py_XDECREF(_traceback);
         return NULL;
```
Comment (Starbuck5) [inline src_c/music.c:391]: I believe this whole block can be removed, since pgRWops_FromObject will set the FileNotFounderror.

Just spotted that :)

**Review by Starbuck5 — APPROVED** [review]
Thanks!

**Pull request merged**